app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1153,11 +1154,36 @@ public void listOptionMenuItemClicked() {
 
     @Override
     public void populatePlaces(final fr.free.nrw.commons.location.LatLng curlatLng) {
-        if (curlatLng.equals(getLastMapFocus())) { // Means we are checking around current location
-            populatePlacesForCurrentLocation(getLastMapFocus(), curlatLng, null);
-        } else {
-            populatePlacesForAnotherLocation(getLastMapFocus(), curlatLng, null);
+        IGeoPoint screenTopRight = mapView.getProjection().fromPixels(mapView.getWidth(), 0);
+        IGeoPoint screenBottomLeft = mapView.getProjection().fromPixels(0, mapView.getHeight());
+        fr.free.nrw.commons.location.LatLng screenTopRightLatLng = new fr.free.nrw.commons.location.LatLng(screenBottomLeft.getLatitude(),screenBottomLeft.getLongitude(),0);
+        fr.free.nrw.commons.location.LatLng screenBottomLeftLatLng = new fr.free.nrw.commons.location.LatLng(screenTopRight.getLatitude(),screenTopRight.getLongitude(),0);
+        fr.free.nrw.commons.location.LatLng emptyLatLng = new fr.free.nrw.commons.location.LatLng(0.0,0.0,0);
+
+        if (screenTopRightLatLng.getLatitude() == 0.0 && screenTopRightLatLng.getLongitude() == 0.0 && screenBottomLeftLatLng.getLatitude() == 0.0 && screenBottomLeftLatLng.getLongitude() == 0.0){
```

Comment:
Can you add a comment explaining in which case that would typically happen?
Also, comparing doubles can be tricky, see https://stackoverflow.com/a/25160509/226958

Reply:
Sure, I have added documentation in newer commit https://github.com/commons-app/apps-android-commons/pull/5553/commits/4d9b10b90536d56e83e820ede39b01ec19aa5577